bin/grommet.js

```diff
@@ -17,76 +18,159 @@ String.prototype.capitalize = function() {
 
-    mkdirp('./' + app, function(err) {
-      if (err) {
-        console.log('Error trying to create project: ' + err);
-      } else {
-        process.chdir('./' + app);
-
-        var templateFolder = path.join(grommetPath, 'templates/init/**');
-        var mobileIcon = path.join(grommetPath, 'img/mobile-app-icon.png');
-        var shortcutIcon = path.join(grommetPath, 'img/shortcut-icon.png');
-
-        gulp.src(mobileIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(templateFolder, { dot: true })
-        .pipe(template({
-          appName: app,
-          appTitle: title,
-          grommetVersion: packageJSON.version
-        }))
-        .pipe(gulp.dest('./'))
-        .pipe(install()).on('finish', function() {
-          done();
-        });
+      if (fileExists(newAppPath)) {
+        console.error('[grommet] Error while creating new app. Directory "'+command.options.app+'" already exists.');
+        process.exit(1);
       }
-    });
+
+      mkdirp(newAppPath, function(err) {
+        if (err) {
+          console.error('[grommet] Error trying to create project: ' + err);
+          process.exit(1);
+        }
+
+        try {
+          var templateFolder = path.join(command.env.templatePath, 'new/**');
+          copyAssetsAndInstall(templateFolder, newAppPath, command);
+        } catch(err) {
+          shelljs.rm('-rf', newAppPath);
+          console.error('[grommet] Error while creating new app '+command.options.app+'.');
+          throw err;
+        }
+      })
+    }
   }
 };
 
+var supportedNodeVersion = '0.10';
+var supportedNpmVersion = '1.4';
 var grommetPath = path.join(__dirname, '..');
-var packageJSON = require(path.join(grommetPath, 'package.json'));
-var npmVersion = Number(shelljs.exec('npm --version', {silent:true})
-  .stdout.toString().match(/^(\d+\.\d+)/)[1]);
 var argv = require('yargs').boolean('version').argv;
+var options = argv._;
+var command = {
+  task: options[0] || 'init',
+  env: {
+    grommetPath: grommetPath,
+    grommetVersion: require(path.join(grommetPath, 'package.json')).version,
+    templatePath: path.join(grommetPath, 'templates'),
+    npmVersion: Number(shelljs.exec('npm --version', {silent:true}).stdout.toString().match(/^(\d+\.\d+)/)[1]),
+    nodeVersion: Number(process.version.match(/^v(\d+\.\d+)/)[1]),
+    processVersion: process.version
+  },
+  options: {}
+}
 
 if (argv.version) {
-  console.log(packageJSON.version);
+  console.log(command.env.grommetVersion);
   process.exit(0);
 }
 
-var options = argv._;
-var command = options[0] || 'init';
-if (command === 'init') {
-  var app = options[1] || 'app-name';
-  var title = app.replace(/-|_/g, ' ').capitalize();
+switch(command.task) {
+  case 'init':
+    command.options['app'] = options[1] || 'app-name';
+    command.options['title'] = command.options.app.replace(/-|_/g, ' ').capitalize();
+    break;
+  case 'new':
```

Comment:
> I have a use case that is coming up soon that is to generate a new project to be published in github (static site).
> 
> I think we could deprecate init and add one more attribute to new so that we could do:
> 
> grommet new empty sample-app "Some description"
> grommet new full sample-app "Some description"
> grommet new static sample-app "Some description"

Are *empty*, *full* and *static* some sort of template? If yes, I think it should be designed as an optional argument, as follows:

```
$ grommet new [-t <template-name>] <app-name> [<description>] 
```

In any case, don't you think this new change should be part of another pull request?